pyuvm/reg/uvm_mem.py

```diff
@@ -0,0 +1,486 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, ClassVar
+
+from pyuvm.error_classes import UVMFatalError
+from pyuvm.reg.uvm_reg_model import (
+    uvm_coverage_model_e,
+    uvm_door_e,
+)
+from pyuvm.s05_base_classes import uvm_object
+
+if TYPE_CHECKING:
+    from pyuvm.reg.uvm_mem_mam import (
+        alloc_mode_e,
+        locality_e,
+        uvm_mem_mam,
+        uvm_mem_mam_cfg,
+    )
+    from pyuvm.reg.uvm_reg_backdoor import uvm_reg_backdoor
+    from pyuvm.reg.uvm_reg_block import uvm_reg_block
+    from pyuvm.reg.uvm_reg_item import uvm_reg_item
+    from pyuvm.reg.uvm_reg_map import uvm_reg_map, uvm_reg_map_info
+    from pyuvm.reg.uvm_reg_model import (
+        uvm_hdl_path_concat,
+        uvm_hdl_path_slice,
+        uvm_reg_addr_t,
+        uvm_reg_cvr_t,
+        uvm_reg_data_t,
+        uvm_status_e,
+    )
+    from pyuvm.reg.uvm_reg_sequence import uvm_reg_frontdoor
+    from pyuvm.reg.uvm_vreg import uvm_vreg
+    from pyuvm.reg.uvm_vreg_field import uvm_vreg_field
+    from pyuvm.s14_15_python_sequences import uvm_sequence_base
+
+__all__ = ["uvm_mem"]
+logger = logging.getLogger("RegModel")
```

Comment:
Does UVM have the concept of a canonical name for components in the hierarchy? It might be a good idea to have each instance have its own logger based on that name.

Feel free to ignore this for now. This can be follow-on.